openssl/src/x509/mod.rs

```diff
@@ -1472,6 +1511,46 @@ impl X509RevokedRef {
             Asn1IntegerRef::from_ptr(r as *mut _)
         }
     }
+
+    /// Get the issuer name of the revoked certificate
+    #[corresponds(X509_REVOKED_get_ext_d2i)]
+    pub fn issuer_name(&self) -> Option<Stack<GeneralName>> {
```

Comment:
In keeping with the general attempt to have 1:1 bindings to OpenSSL, how about switching to a single generic method that retrieves extensions? Something like:

```rust
impl X509RevokedRef {
    pub fn extension<T>(&self) -> Option<T::Output>
        where T: X509Extension
    { ... }
}

pub unsafe trait X509Extension {
    const NID: Nid;

    type Output: ForeignType;
}

pub enum CertificateIssuer {}

unsafe impl X509Extension for CertificateIssuer {
    const NID: Nid = Nid::CERTIFICATE_ISSUER;

    type Output = Stack<GeneralName>;
}
```

Reply:
I've swapped over to this, I think it's a nicer match for the OpenSSL API and we should probably do the same thing to all the others but it is definitely a bit more complicated. (I did also deal with the crit parameter correctly at the same time which didn't help).